monai/transforms/utility/dictionary.py

```diff
@@ -425,7 +425,13 @@ def __call__(self, data: Mapping[Hashable, NdarrayOrTensor]) -> Dict[Hashable, N
                         d[split_meta_key] = deepcopy(orig_meta)
                         dim = self.splitter.dim
                         if dim > 0:  # don't update affine if channel dim
-                            shift = np.eye(len(d[split_meta_key]["affine"]))  # type: ignore
+                            affine = d[split_meta_key]["affine"]  # type: ignore
+                            ndim = len(affine)
+                            shift = (
```

Comment:
Maybe a little clearer as statements:

```suggestion
if isinstance(affine, torch.Tensor):
    shift = torch.eye(ndim, device=affine.device, dtype=affine.dtype)
else:
    shift = np.eye(ndim)
```